knockout-file-bind.js

```diff
@@ -21,67 +21,78 @@
-      // event without actually picking a file.
-      if (file === undefined) {
-        fileContents(null)
-        return;
-      }
-
-      if (allowed) {
-        if (!allowed.some(function(type) { return type === file.type })) {
-          console.log("File "+file.name+" is not an allowed type, ignoring.")
-          fileContents(null)
-          return;
+    ko.bindingHandlers['file'] = {
+      init: function(element, valueAccessor, allBindings) {
+        var fileContents, fileName, allowed, prohibited, reader;
+    
+        if ((typeof valueAccessor()) === "function") {
+          fileContents = valueAccessor();
+        } else {
+          fileContents = valueAccessor()['data'];
+          fileName = valueAccessor()['name'];
+    
+          allowed = valueAccessor()['allowed'];
+          if ((typeof allowed) === 'string') {
+            allowed = [allowed];
+          }
+    
+          prohibited = valueAccessor()['prohibited'];
+          if ((typeof prohibited) === 'string') {
+            prohibited = [prohibited];
+          }
+    
+          reader = (valueAccessor()['reader']);
         }
-      }
-
-      if (prohibited) {
-        if (prohibited.some(function(type) { return type === file.type })) {
-          console.log("File "+file.name+" is a prohibited type, ignoring.")
-          fileContents(null)
-          return;
+    
+        reader || (reader = new FileReader());
+        reader.onloadend = function() {
+          fileContents(reader.result);
         }
-      }
-
-      reader.readAsDataURL(file); // A callback (above) will set fileContents
-      if (typeof fileName === "function") {
-        fileName(file.name)
+    
+        var handler = function() {
+          var file = element.files[0];
+    
+          // Opening the file picker then canceling will trigger a 'change'
+          // event without actually picking a file.
+          if (file === undefined) {
+            fileContents(null)
+            return;
+          }
+    
+          if (allowed) {
+            if (!allowed.some(function(type) { return type === file.type })) {
+              console.log("File "+file.name+" is not an allowed type, ignoring.")
+              fileContents(null)
+              return;
+            }
+          }
+    
+          if (prohibited) {
+            if (prohibited.some(function(type) { return type === file.type })) {
+              console.log("File "+file.name+" is a prohibited type, ignoring.")
+              fileContents(null)
+              return;
+            }
+          }
+    
+          reader.readAsDataURL(file); // A callback (above) will set fileContents
+          if (typeof fileName === "function") {
+            fileName(file.name)
+          }
+        }
+    
+        ko.utils.registerEventHandler(element, 'change', handler);
       }
     }
+  }
 
-    ko.utils.registerEventHandler(element, 'change', handler);
+  if (define) {
```

Comment:
if not present it would be `undefined`, which I assume would result to `false` in a boolean condition.  The values `false`, `null`, `0` and `undefined` are treated all as `false` in if-else logic.

If you have observed otherwise, I will change that line to something more meaningful. For example `if (define && typeof define === 'function')` is indeed more accurate.